tutorials/haskell/snap-web-app/tutorial.md

```diff
@@ -0,0 +1,490 @@
+---
+title: How to create a web application in Haskell with Snap
+published: 2017-01-30
+ghc: 8.0.1
+lts: 7.18
+libraries: snap-core-1.0.1.0 snap-server-1.0.1.1 heist-1.0.1.0 HDBC-sqlite3-2.3.3.1    
+language: haskell
+description: In this tutorial we show you how to build a simple TODO web app using Snap framework
+author-name: Omar García
+---
+
+## Introduction
+
+[Snap](http://snapframework.com/) is a simple and [fast](http://snapframework.com/blog/2010/11/17/snap-0.3-benchmarks)
+web development framework for the Haskell programming language with a clean
```

Comment:
... Snap web applications are composed of snaplets, which are self-contained pieces of reusable functionality that can be glued together. ...